tests/integration/conftest.py

```diff
@@ -1,32 +1,151 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+"""Fixtures for github runner charm integration tests."""
+
+import secrets
+import subprocess
 from pathlib import Path
+from typing import Any, AsyncIterator
 
 import pytest
+import pytest_asyncio
 import yaml
+from juju.application import Application
+from juju.model import Model
+from pytest_operator.plugin import OpsTest
 
 
-@pytest.fixture
-def metadata():
+@pytest.fixture(scope="module")
+def metadata() -> dict[str, Any]:
+    """Metadata information of the charm"""
     metadata = Path("./metadata.yaml")
     data = yaml.safe_load(metadata.read_text())
     return data
 
 
-@pytest.fixture
-def model(ops_test):
+@pytest.fixture(scope="module")
+def app_name() -> str:
+    # Randomized app name to avoid collision when connecting to GitHub.
+    return f"integration-{secrets.token_hex(4)}"
+
+
+@pytest.fixture(scope="module")
+def path(pytestconfig: pytest.Config) -> str:
+    path = pytestconfig.getoption("--path")
+    assert path, "Please specify the --path command line option"
+    return path
+
+
+@pytest.fixture(scope="module")
+def token(pytestconfig: pytest.Config) -> str:
+    token = pytestconfig.getoption("--token")
+    assert token, "Please specify the --token command line option"
+    return token
+
+
+@pytest.fixture(scope="module")
+def token_alt(pytestconfig: pytest.Config, token: str) -> str:
+    token_alt = pytestconfig.getoption("--token-alt")
+    assert token_alt, "Please specify the --token-alt command line option"
+    assert token_alt != token, "Please specify a different token for --token-alt"
+    return token_alt
+
+
+@pytest.fixture(scope="module")
+def http_proxy(pytestconfig: pytest.Config) -> str:
+    http_proxy = pytestconfig.getoption("--http-proxy")
+    return "" if http_proxy is None else http_proxy
+
+
+@pytest.fixture(scope="module")
+def https_proxy(pytestconfig: pytest.Config) -> str:
+    https_proxy = pytestconfig.getoption("--https-proxy")
+    return "" if https_proxy is None else https_proxy
+
+
+@pytest.fixture(scope="module")
+def no_proxy(pytestconfig: pytest.Config) -> str:
+    no_proxy = pytestconfig.getoption("--no-proxy")
+    return "" if no_proxy is None else no_proxy
+
+
+@pytest.fixture(scope="module")
+def model(ops_test: OpsTest) -> Model:
+    assert ops_test.model is not None
     return ops_test.model
 
 
-@pytest.fixture
-def application(model, metadata):
-    charm_name = metadata["name"]
-    app = model.applications[charm_name]
-    return app
+@pytest_asyncio.fixture(scope="module")
+async def lxd_profile() -> AsyncIterator[Path]:
+    lxd_profile_path = Path("lxd-profile.yaml")
+
+    with open(lxd_profile_path, "w") as profile_file:
```

Comment:
Minor nitpick, I think tempfile could also be used so that the following unlink can be removed.

Reply:
This file needs to be have this exact name under this path. I think it is not possible with the `tempfile` module.